### PR TITLE
Added Google Cloud Run tutorial

### PR DESCRIPTION
**Pull request by ajeetraina**

### CODE REVIEW

File: docs/create/cloudrun/index-cloudrun.mdx [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+authors: [ajeet]
+---
+
+
+![alt_text](images/cloudrun_logo.png "image_tooltip")
+
+
+If you’re looking for a solution that lets you go from container to URL within seconds, check out Google Cloud Run. [Cloud Run](https://cloud.google.com/run) is Google’s fully managed compute platform for running stateless, HTTP-driven containers. By using a single command (“[gcloud run deploy”](https://cloud.google.com/sdk/gcloud/reference/run/deploy)) you can convert a container image to a fully managed web application and run it in a production environment with auto-scalability, high availability, and security.
+
+[Google Cloud Run](https://cloud.google.com/run/docs) allows you to deploy and scale serverless HTTP containers without worrying about provisioning servers, scaling servers up and down to meet demands, or overpaying when you are allocated with more resources than necessary. It makes container deployment much easier. It is good for developing software in cloud applications, hence delivering web apps, [API,](https://cloud.google.com/run/docs/reference/rest) background jobs, etc.
```
Comment (simonprickett) [inline docs/create/cloudrun/index-cloudrun.mdx]:
```suggestion
[Google Cloud Run](https://cloud.google.com/run/docs) allows you to deploy and scale serverless HTTP containers without worrying about provisioning servers, scaling servers up and down to meet demands, or overpaying by consuming more resources than necessary. It makes container deployment much easier. It is good for developing software in cloud applications, hence delivering web apps, [APIs](https://cloud.google.com/run/docs/reference/rest), background jobs, etc.
```

File: docs/create/cloudrun/index-cloudrun.mdx [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+
+Cloud Run is [powered and built on Knative](https://cloud.google.com/knative). Knative is an open source community project which adds components for deploying, running, and managing serverless, cloud-native applications to Kubernetes. It allows you to easily run your containers either in your [https://cloud.google.com/kubernetes-engine](https://cloud.google.com/kubernetes-engine/) (GKE) cluster with Cloud Run on GKE or fully managed with Cloud Run. Cloud Run helps developers focus on writing high-value code, regardless of where their organizations are on the path to the cloud. 
+
+
+### Compelling feature of Google Cloud Run
```
Comment (simonprickett) [inline docs/create/cloudrun/index-cloudrun.mdx]:
```suggestion
### Compelling features of Google Cloud Run
```

File: docs/create/cloudrun/index-cloudrun.mdx [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+
+
+
+- Cloud Run provides you with the option to write your code in your preferred programming languages such as [NodeJS,](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/master/run/helloworld/) [Python](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/run/), [Go](https://github.com/GoogleCloudPlatform/golang-samples/tree/master/run/), and [Java](https://github.com/GoogleCloudPlatform/java-docs-samples/tree/master/run/).
```
Comment (simonprickett) [inline docs/create/cloudrun/index-cloudrun.mdx]:
```suggestion
- Cloud Run provides you with the option to write your code in your preferred programming languages such as [NodeJS](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/master/run/helloworld/), [Python](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/run/), [Go](https://github.com/GoogleCloudPlatform/golang-samples/tree/master/run/), and [Java](https://github.com/GoogleCloudPlatform/java-docs-samples/tree/master/run/).
```

File: docs/create/cloudrun/index-cloudrun.mdx [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+- Cloud Run provides you a better ROI as you get charged only for the resources used.
+- Cloud Run offers a[ longer request timeout duration of up to 60 minutes](https://cloud.google.com/run/docs/configuring/request-timeout).
+- Cloud Run is configured to [send multiple concurrent requests on each container instance. ](https://cloud.google.com/run/docs/configuring/concurrency)This is helpful to improve latency and reduce costs if you’re expecting large volumes.
+- Cloud Run lets you [create microservice-based applications that are scalable and extensible](https://cloud.google.com/blog/products/serverless/build-event-driven-applications-in-cloud-run).
+
+In this tutorial, you will learn how to deploy a simple Redis Rating limiting application to Google Cloud Run in just 5 minutes.
```
Comment (simonprickett) [inline docs/create/cloudrun/index-cloudrun.mdx]:
```suggestion
In this tutorial, you will learn how to deploy a simple Redis rate limiting application to Google Cloud Run in just 5 minutes.
```

File: docs/create/cloudrun/index-cloudrun.mdx [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+* Step 3. Authenticate your GCP account
+* Step 4. Enable Google Services
+* Step 5. Deploy to Google Cloud Run from source
+* Step 6. Verify if service is listed
+* Step 7. Set up environment variables
+* Step 8. Access your app
+
+### Step 1. Set up a free Redis Enterprise Cloud account
+
+Visit [developer.redis.com/create/rediscloud/](https://developer.redis.com/create/rediscloud/) and create [a free Redis Enterprise Cloud account](https://redis.com/try-free/). Once you complete the signup tutorial, you will be provided with the database endpoint URL and password. Save it for future reference.
```
Comment (simonprickett) [inline docs/create/cloudrun/index-cloudrun.mdx]:
```suggestion
Visit [developer.redis.com/create/rediscloud/](https://developer.redis.com/create/rediscloud/) and create [a free Redis Enterprise Cloud account](https://redis.com/try-free/). Once you complete the signup tutorial, you will be provided with the database endpoint URL, port and password. Save these for future reference.
```

File: docs/create/cloudrun/index-cloudrun.mdx [hunk 6]
```diff
@@ -0,0 +1,282 @@
...
+
+
+
+
+![alt_text](images/rediscloud_register.png "image_tooltip")
```
Comment (simonprickett) [inline docs/create/cloudrun/index-cloudrun.mdx]:
```suggestion
![Registering for Redis Cloud](images/rediscloud_register.png "image_tooltip")
```

File: docs/create/cloudrun/index-cloudrun.mdx [hunk 7]
```diff
@@ -0,0 +1,282 @@
...
+
+
+### Step 2. Install Google Cloud CLI
+
+Run the below command to install Google Cloud CLI Core Libraries and dependencies
```
Comment (simonprickett) [inline docs/create/cloudrun/index-cloudrun.mdx]:
```suggestion
Run the following command to install Google Cloud CLI Core Libraries and dependencies:
```

File: docs/create/cloudrun/index-cloudrun.mdx [hunk 8]
```diff
@@ -0,0 +1,282 @@
...
+
+```
+brew install --cask google-cloud-sdk
+
+```
+
+
+### Step 3. Authenticate your GCP account
+
+To deploy your app with the gcloud CLI, you must download, install, and initialize the [gcloud CLI](https://cloud.google.com/sdk/docs).
```
Comment (simonprickett) [inline docs/create/cloudrun/index-cloudrun.mdx]:
```suggestion
To deploy your app you must first download, install, and initialize the [gcloud CLI](https://cloud.google.com/sdk/docs). 
```

File: docs/create/cloudrun/index-cloudrun.mdx [hunk 9]
```diff
@@ -0,0 +1,282 @@
...
+```
+
+
+Allow Google Cloud SDK access to your Google Account:
+
+
+
+
+
+![alt_text](images/authenticate_gcloud.png "image_tooltip")
```
Comment (simonprickett) [inline docs/create/cloudrun/index-cloudrun.mdx]:
```suggestion
![Authenticating with Google Cloud](images/authenticate_gcloud.png "image_tooltip")
```

File: docs/create/cloudrun/index-cloudrun.mdx [hunk 10]
```diff
@@ -0,0 +1,282 @@
...
+```
+
+
+
+### Step 6. Verify if the service is listed under Cloud Run dashboard
+
+
+
+
+![alt_text](images/gcloud1.png "image_tooltip")
```
Comment (simonprickett) [inline docs/create/cloudrun/index-cloudrun.mdx]:
```suggestion
![Verifying if the service is listed](images/gcloud1.png "image_tooltip")
```

File: docs/create/cloudrun/index-cloudrun.mdx [hunk 11]
```diff
@@ -0,0 +1,282 @@
...
+
+
+
+![alt_text](images/gcloud2.png "image_tooltip")
```
Comment (simonprickett) [inline docs/create/cloudrun/index-cloudrun.mdx]:
```suggestion
![Verifying if the service is listed](images/gcloud2.png "image_tooltip")
```

File: docs/create/cloudrun/index-cloudrun.mdx [hunk 12]
```diff
@@ -0,0 +1,282 @@
...
+### Step 7. Set up environment variables 
+
+
+
+- Adding Redis URL to point to Redis Enterprise Cloud
+
+
+
+
+![alt_text](images/gcloud3.png "image_tooltip")
```
Comment (simonprickett) [inline docs/create/cloudrun/index-cloudrun.mdx]:
```suggestion
![Setting up environment variables](images/gcloud3.png "image_tooltip")
```

File: docs/create/cloudrun/index-cloudrun.mdx [hunk 13]
```diff
@@ -0,0 +1,282 @@
...
+
+
+
+
+- Enable HTTPS
+
+
+
+
+![alt_text](images/gcloud4.png "image_tooltip")
```
Comment (simonprickett) [inline docs/create/cloudrun/index-cloudrun.mdx]:
```suggestion
![Enabling HTTPS](images/gcloud4.png "image_tooltip")
```

File: docs/create/cloudrun/index-cloudrun.mdx [hunk 14]
```diff
@@ -0,0 +1,282 @@
...
+
+
+
+-  Ensuring the VPC connector shows as “none”
+
+
+
+![alt_text](images/gcloud5.png "image_tooltip")
```
Comment (simonprickett) [inline docs/create/cloudrun/index-cloudrun.mdx]:
```suggestion
![Configuring the VPC connector](images/gcloud5.png "image_tooltip")
```

File: docs/create/cloudrun/index-cloudrun.mdx [hunk 15]
```diff
@@ -0,0 +1,282 @@
...
+
+
+
+- Allowing all traffic
+
+
+
+![alt_text](images/gcloud6.png "image_tooltip")
```
Comment (simonprickett) [inline docs/create/cloudrun/index-cloudrun.mdx]:
```suggestion
![Allowing all traffic](images/gcloud6.png "image_tooltip")
```

File: docs/create/cloudrun/index-cloudrun.mdx [hunk 16]
```diff
@@ -0,0 +1,282 @@
...
+
+
+### Step 8. Access your app
+
+Open [https://django-redis-rate-limiting-example-opcboau66a-de.a.run.app/](https://django-redis-rate-limiting-example-opcboau66a-de.a.run.app/) and access your app.
```
Comment (simonprickett) [inline docs/create/cloudrun/index-cloudrun.mdx:270]: @ajeetraina maybe needs a sentence or two explaining where this URL comes from and that the URL in the article here is just an example, your URL will differ if you are following this tutorial?

File: docs/create/cloudrun/index-cloudrun.mdx [hunk 17]
```diff
@@ -0,0 +1,282 @@
...
+
+
+
+![alt_text](images/ratelimit.png "image_tooltip")
```
Comment (simonprickett) [inline docs/create/cloudrun/index-cloudrun.mdx]:
```suggestion
![Running the application](images/ratelimit.png "image_tooltip")
```

File: docs/create/cloudrun/index-cloudrun.mdx [hunk 18]
```diff
@@ -0,0 +1,282 @@
...
+
+
+This app allows you to choose the desired requests every 10 seconds. The app blocks connections from a client after surpassing a certain amount of requests (default: 10) per time (default: 10 sec). That will let the user know how many requests they have remaining before they run over the limit. On the tenth run, the server should return an HTTP status code of 429 (“Too Many Requests”).
```
Comment (simonprickett) [inline docs/create/cloudrun/index-cloudrun.mdx]:
```suggestion
This app allows you to choose the desired number of requests in each 10 second period. The app blocks connections from a client after surpassing a certain amount of requests (default: 10) in the time window (default: 10 sec). That will let the user know how many requests they have remaining before they run over the limit. On the tenth run, the server should return an HTTP status code of 429 (“Too Many Requests”).
```

**Review by simonprickett — APPROVED** [review]
@ajeetraina looks good - left a few small comments for consideration.

File: docs/create/cloudrun/index-cloudrun.mdx (hunk 16 above)
Comment (ajeetraina) [reply docs/create/cloudrun/index-cloudrun.mdx:270]: Let me modify it accordingly.

**Pull request merged**